File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        if k == len(nums):
            return nums
        count = defaultdict(int)
        ans = []

        for num in nums:
            count[num] += 1
        
        freq = []

        for i in range(len(nums)+1):
            freq.append([])

        
        for num, cnt in count.items():
            freq[cnt].append(num)

        for i in reversed(range(len(freq))):
            for j in range(len(freq[i])):
                ans.append(freq[i][j])
                if k == len(ans):
                    return ans

File: Data Structures & Algorithms/majority-element/submission-4.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0;
        int curr = 0;

        for (int i=0; i<nums.size(); i++) {
            if (count == 0) {
                curr = nums[i];
            }

            if (nums[i] == curr) count++;
            if (nums[i] != curr) count--;
        }        

        return curr;
    }
};